### ISSUE TITLE
Feature request: enable environment casing inside a settings yaml file

### ISSUE THREAD

**Issue by ehudyalinmor** [railsconfig/config#105]
Sometimes splitting environment configuration to different files can be very hard to maintain. 
For instance, if you are using the settings to configure an external service such as bit.ly you will naturally have:

```yml
bitly.yml
  bitly:
    threshold: 1000
    protocol: https
bitly.development.yml
  bitly:
    login: "foo"
    api_key: "bar"

bitly.production.yml
  bitly:
    login: "foofoo"
    api_key: "barbar"
```

When having a lot of environment dependant configurations and some other non-environment dependant configurations makes the configuration hard to follow, read and maintain.

My suggestion is to introduce a new syntax when you can define environment case inside the one config file. For example, a cleaner representation of the bit.ly configuraiton above will be:

```yml
bitly.yml
  bitly:
    threshold: 1000
    protocol: https

    development:
      login: "foo"
      api_key: "xyz"

    production:
      login: "bar"
      api_key: "abc"
```

It required to predefine what environments are available but it really simplifies the config files when there are a lot of configurations.

**Comment by pkuczynski** [comment]
Why don't you store your `bitly` configuration together with your other settings?

```yml
default.yml
  bitly:
    threshold: 1000
    protocol: https
 other_setting:
    setting: 100

development.yml
  bitly:
    login: "foo"
    api_key: "bar"
 other_setting:
    setting: 200

bitly.production.yml
  bitly:
    login: "foofoo"
    api_key: "barbar"
 other_setting:
    setting: 300
```

**Comment by pkuczynski** [comment]
I see what you mean, but this is against the "split config into separate env-based files" idea which was the merit for this gem. You should store all common and default bitty settings in `config/settings.yml` file and override only some of them when needed for a particular env.
